src/parser/ParsedQuery.h

```diff
@@ -157,6 +94,10 @@ class ParsedQuery {
     return std::holds_alternative<ConstructClause>(_clause);
   }
 
+  [[nodiscard]] bool hasUpdateClause() const {
+    return std::holds_alternative<UpdateClause>(_clause);
+  }
+
```

Comment:
What happens with this when you type an Update?
Will you get a cryptic AD_FAIL() or even the raw exception of `std::get` because code implicitly assumes that not SELECT means CONSTRUCT?

I think an easy way for now would be to at some point (e.g. at the end of the Parser, in one of the highest rules) 
Let all the Updates fail. Maybe I overlooked it, but I am sure you will tell me.

Reply:
If one were to pass an update `ParsedQuery` to the server it would fail with a `std::get` error like you mentioned. The `queryOrUpdate` rule still directly fails.